beacon-chain/rpc/beaconv1/pool.go

```diff
@@ -22,8 +23,40 @@ func (bs *Server) ListPoolAttestations(ctx context.Context, req *ethpb.Attestati
 
 // SubmitAttestation submits Attestation object to node. If attestation passes all validation
 // constraints, node MUST publish attestation on appropriate subnet.
-func (bs *Server) SubmitAttestation(ctx context.Context, req *ethpb.Attestation) (*ptypes.Empty, error) {
-	return nil, errors.New("unimplemented")
+func (bs *Server) SubmitAttestations(ctx context.Context, req *ethpb.SubmitAttestationsRequest) (*ptypes.Empty, error) {
+	ctx, span := trace.StartSpan(ctx, "beaconv1.SubmitAttestation")
+	defer span.End()
+
+	headState, err := bs.ChainInfoFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+
+	var validAttestations []*ethpb_alpha.Attestation
+	for _, sourceAtt := range req.Data {
+		att := migration.V1AttToV1Alpha1(sourceAtt)
+		err = blocks.VerifyAttestationSignature(ctx, headState, att)
```

Comment:
According to the function comment `If attestation passes all validation constraints...`, just verifying the signature is not enough here